### ISSUE TITLE
[BUG]: Azure App service deploy - error

### ISSUE THREAD

**Issue by roofiq** [microsoft/azure-pipelines-tasks#18361]
### Task name

AzureRmWebAppDeployment@4

### Task version

4.223.0

### Environment type (Please select at least one enviroment where you face this issue)

- [X] Self-Hosted
- [ ] Microsoft Hosted
- [ ] VMSS Pool
- [ ] Container

### Azure DevOps Server type

dev.azure.com (formerly visualstudio.com)

### Operation system

Windows 10

### Task log

```shell
##[debug]Processed: ##vso[telemetry.publish area=TaskDeploymentMethod;feature=AzureWebAppDeployment]{"deploymentMethod":"Web Deploy"}
##[debug]Performing the deployment of webapp.
##[debug]Could not parse response: {}
##[debug]Response: undefined
##[debug]Correlation ID from ARM api call response : 82066c44-201c-4e93-ae9c-2c3e2afdc877
##[debug]Deployment Failed with Error: TypeError: Cannot read property 'retrieveSecret' of undefined
##[debug]task result: Failed
##[error]TypeError: Cannot read property 'retrieveSecret' of undefined
##[debug]Processed: ##vso[task.issue type=error;]TypeError: Cannot read property 'retrieveSecret' of undefined
##[debug]Processed: ##vso[task.complete result=Failed;]TypeError: Cannot read property 'retrieveSecret' of undefined
##[debug][GET]https://management.azure.com/subscriptions/c0a9f8c5-ba51-48e8-b1b8-a9e36cef6220/providers/microsoft.insights/components?$filter=InstrumentationKey eq '***'&api-version=2015-05-01
##[debug]Correlation ID from ARM api call response : 47b3891c-e358-4921-b000-d40e01dcd592
##[debug]Unable to find Application Insights resource with Instrumentation key ***. Skipping adding release annotation.
##[debug]Azure_App_Service_Deploy_PackageArtifactAlias=
##[debug]release.releaseId=undefined
##[debug]release.releaseName=undefined
```

### Relevant log output

```shell
##[error]TypeError: Cannot read property 'retrieveSecret' of undefined
```

### Aditional info

```shell
Hello,

even today morning, when using versiom 4.221.103, the task was finishing succesfully. But few minutes ago, our pipelines started failing, with the error I added above. The visible version for the task is 4.223.0

Thanks for the support.
```

**Comment by AlphaSolutions-jbj** [comment]
Exact same issue here.

**Comment by sorted-software** [comment]
Experiencing this issue on our Pipeline too.  Europe West.

**Comment by TitaniumIT** [comment]
my too

**Comment by janssensjeroen** [comment]
Same issue here as well: Europe West, self-hosted, Windows.

**Comment by korvatap** [comment]
We are experiencing this issue in our Pipeline also—Europe West: Azure Hosted agent, windows-latest.

**Comment by LauraXiulan** [comment]
Same issue here in the release pipeline. Europe West, self-hosted, Windows.

**Comment by chrisdecker1201** [comment]
We have the same issue here, too.

**Comment by FabrizioSalis** [comment]
It seems to be working now. Thanks.

**Comment by zulkiflifauzi** [comment]
confirmed, mine is working now. Thanks!

**Comment by oakular** [comment]
Working for us now, thank you.

**Comment by jferrante-ws** [comment]
We are still experiencing this error, is it still in the process of being deployed across all instances/regions of azure devops?

**Comment by aquarelacs-arthur-rosa** [comment]
Could you tell the regions that are still experiencing this error?

**Comment by jferrante-ws** [comment]
@aquarelacs-arthur-rosa the region is "Central US"

**Comment by keipa** [comment]
![image](https://github.com/microsoft/azure-pipelines-tasks/assets/11652476/c2269ac3-7eac-49fc-99e0-dca0a8149d98)
solution, just switch to version 3

**Comment by jferrante-ws** [comment]
@keipa that appears to be a temporary work around?  Reading above it sounds like some regions actually have a fix vs having to utilize any workarounds.

**Comment by redcloud319** [comment]
Im also still experiencing issues with Task using Version 4.* . Currently the Agent Pool that i am using is 'Windows-lastest' and deploying to Azure WestUS. I dont necessarily want to downgrade my pipeline temporarily. How long will it take for Agents to fall back to the other Version?

**Comment by keipa** [comment]
**Don't forget to review parameters if you downgrade your version**

**Comment by jferrante-ws** [comment]
Deploys working again for us in Central US.

**Comment by ozanaydinMSFT** [comment]
New fix version `4.223.1` is on the way for the still affected regions, it will arrive within an hour.
